<root BTCPP_format="4" >
    <BehaviorTree ID="MainTree">
        <Sequence name="root_sequence"> 
            <Sequence name="RobotInitializer">
            <RobotInitializer name="RobotInitializer" message="RobotInitializer"/>
            <ChangeGripper name="open_gripper" position="0.3" max_effort = "0.0"/>
            </Sequence>
            
            <Sequence name="pickup_sensor_from_palette">
            <Pregrasp name="pickup_sensor_from_palette" action="pickup_sensor_from_palette" target_x="-0.1" target_y="0.01" target_z="0.8" target_roll="0.01" target_pitch ="3.14" target_yaw = "0.01" pregrasp_offset="0.1"/>
            <ChangeGripper name="close_gripper" position="0.6" max_effort = "0.0"/>
            </Sequence>

            <Sequence name="deploy_sensor">
            <Pregrasp name="deploy_sensor" action="deploy_sensor" target_x="0.6" target_y="0.01" target_z="0.5" target_roll="0.01" target_pitch ="3.14" target_yaw = "0.01" pregrasp_offset="0.1"/>
            <ChangeGripper name="open_gripper" position="0.3" max_effort = "0.0"/>
            </Sequence>

            <Sequence name="restore_arm_pose">
            <RobotInitializer name="restore_arm_pose" message="restore_arm_pose"/>
            </Sequence>

        </Sequence>
    </BehaviorTree>
    <TreeNodesModel>
        <Action ID="ChangeGripper">
            <input_port name="max_effort" type="double"/>
            <input_port name="position" type="double"/>
        </Action>
        <Action ID="Drop"/>
        <Action ID="Grasp">
            <input_port name="target_roll" type="double"/>
            <input_port name="target_pitch" type="double"/>
            <input_port name="target_z" type="double"/>
            <input_port name="target_y" type="double"/>
            <input_port name="target_x" type="double"/>
            <input_port name="target_yaw" type="double"/>
            <input_port name="action" type="std::string"/>
        </Action>
        <Action ID="Pregrasp">
            <input_port name="target_roll" type="double"/>
            <input_port name="target_pitch" type="double"/>
            <input_port name="pregrasp_offset" type="double"/>
            <input_port name="target_z" type="double"/>
            <input_port name="target_y" type="double"/>
            <input_port name="target_x" type="double"/>
            <input_port name="target_yaw" type="double"/>
            <input_port name="action" type="std::string"/>
        </Action>
        <Action ID="RetreatZ"/>
        <Action ID="RobotInitializer">
            <input_port name="message" type="std::string"/>
        </Action>
        <Action ID="ScanPose"/>
    </TreeNodesModel>
</root>